packages/veritone-widgets/src/widgets/TranscriptEngineOutputWidget/index.js

```diff
@@ -0,0 +1,231 @@
+  }),
+  {
+    //undo: TranscriptRedux.undo,           //Uncomment when needed to enable undo option 
+    //redo: TranscriptRedux.redo,           //Uncomment when needed to enable redo option
+    change: changeWidthDebounce,
+    reset: TranscriptRedux.reset,
+    clearData: TranscriptRedux.clearData,
+    receiveData: TranscriptRedux.receiveData
+  },
+  null,
+  { withRef: true }
+)
+export default class TranscriptEngineOutputWidget extends Component {
+  static propTypes = {
+    data: arrayOf(
+      shape({
+        startTimeMs: number,
+        stopTimeMs: number,
+        status: string,
+        series: arrayOf(
+          shape({
+            startTimeMs: number,
+            stopTimeMs: number,
+            words: arrayOf(
+              shape({
+                word: string,
+                confidence: number
+              })
+            )
+          })
+        )
+      })
+    ),
+    currentData: arrayOf(
+      shape({
+        startTimeMs: number,
+        stopTimeMs: number,
+        status: string,
+        series: arrayOf(
+          shape({})
+        )
+      })
+    ),
+
+    selectedEngineId: string,
+    engines: arrayOf(
+      shape({
+        id: string,
+        name: string
+      })
+    ),
+    title: string,
+
+    className: string,
+    headerClassName: string,
+    contentClassName: string,
+
+    editMode: bool,
+
+    onClick: func,
+    onScroll: func,
+    onEngineChange: func,
+    onExpandClicked: func,
+
+    mediaLengthMs: number,
+    neglectableTimeMs: number,
+    estimatedDisplayTimeMs: number,
+
+    mediaPlayerTimeMs: number,
+    mediaPlayerTimeIntervalMs: number,
+
+    //undo: func,     //Uncomment when needed to enable undo option
+    //redo: func,     //Uncomment when needed to enable redo option
+    reset: func.isRequired,
+    change: func.isRequired,
+    clearData: func.isRequired,
+    receiveData: func.isRequired,
+    hasChanged: bool
+  }
+
+  state = {
+    alert: false,
+    editMode: TranscriptEditMode.SNIPPET
+  };
+
+  // use this when we update to react ^16.3.0
+  /*
+  static getDerivedStateFromProps (nextProps, prevState) {
+    const hasNewData = !isEqual(this.props.data, nextProps.data);
+    const hasInitialData = isEqual(this.props.currentData, nextProps.data);
+    (hasNewData || !hasInitialData) && this.receiveData(nextProps.data);
+  }
+  */
+
+  // Use the above function when we update to a later version of react
+  componentWillReceiveProps (nextProps) {
+    const hasNewData = !isEqual(this.props.data, nextProps.data);
+    const missingInitialData = !this.props.currentData || this.props.currentData.length === 0;
+
+    (hasNewData || missingInitialData) && this.props.receiveData(nextProps.data);
+  }
+
+  componentWillUnmount() {
+    this.props.clearData && this.props.clearData();
+  }
+
+  handleContentChanged = (value) => {
+    this.props.change(value);
+  }
+
+  handleOnEditModeChange = (value) => {
+    if (this.props.hasChanged) {
```

Comment:
this can be simplified to `this.setState({ alert: this.props.hasChanged, editMode: value.type })`, no need for the if/else logic

Reply:
ah, one of the is pendingMode and the other is editMode.

Reply:
oh strange, my bad then